src/esp/assets/GenericInstanceMeshData.cpp

```diff
@@ -42,33 +46,30 @@ void copyTo(std::shared_ptr<tinyply::PlyData> data, std::vector<T>& dst) {
 }
 }  // namespace
 
-/*
- * In modern OpenGL fragment sharder, we can access the ID of the current
- * primitive and thus can index an array.  We can make a 2D texture behave
- * like a 2D array with nearest sampling and edge clamping.
- */
-Magnum::GL::Texture2D createInstanceTexture(float* data, const int texSize) {
-  /*
-   * Create the image that wil be uploaded to the texture.  We
-   * can index the original data array
-   * with image[primId / texSize, primId % texSize]
-   * NB: The indices will have to mapped into [0, 1] in the GLSL code
-   */
-  Magnum::Image2D image(
-      Magnum::PixelFormat::R32F, {texSize, texSize},
-      Corrade::Containers::Array<char>(reinterpret_cast<char*>(data),
-                                       texSize * texSize * sizeof(data[0])));
-
-  Magnum::GL::Texture2D tex;
-
-  tex.setMinificationFilter(Magnum::SamplerFilter::Nearest)
-      .setMagnificationFilter(Magnum::SamplerFilter::Nearest)
-      .setWrapping({Magnum::SamplerWrapping::ClampToEdge,
-                    Magnum::SamplerWrapping::ClampToEdge})
-      .setStorage(1, Magnum::GL::TextureFormat::R32F, image.size())
-      .setSubImage(0, {}, image);
-
-  return tex;
+Mn::GL::Texture2D createInstanceTexture(
+    Cr::Containers::ArrayView<std::uint32_t> objectIds) {
+  const Mn::Vector2i textureSize{
+      gfx::PrimitiveIDTexturedShader::PrimitiveIDTextureWidth,
+      int(objectIds.size() +
```

Comment:
```suggestion
      int(std::ceil(float(objectIds.size()) / gfx::PrimitiveIDTexturedShader::PrimitiveIDTextureWidth))
```

Reply:
Well. I'm usually going over code and replacing `ceil(float())` with the pure-integer solution, not the other way around :laughing:

Are you sure this would work correctly for values that have more than ~23 bits of precision? One PLY file has 9M of primitives, which is 24 bits. I'd trust the integer solution there instead ;)

Reply:
I pretty much always prefer `ceil(float())` for readability :) -- The precision definitely isn't an issue here.  The precision just needs to be good enough to separate things by `gfx::PrimitiveIDTexturedShader::PrimitiveIDTextureWidth` (which, since it's a power of 2, can be perfectly represented a float).

Reply:
But this is entirely a personal preference, I am fine with it either way :)